sdk/data/aztables/error_codes.go

```diff
@@ -0,0 +1,107 @@
+type TableErrorCode string
+
+const (
+	DuplicatePropertiesSpecified TableErrorCode = "DuplicatePropertiesSpecified"
+	EntityAlreadyExists          TableErrorCode = "EntityAlreadyExists"
+	EntityNotFound               TableErrorCode = "EntityNotFound"
+	EntityTooLarge               TableErrorCode = "EntityTooLarge"
+	HostInformationNotPresent    TableErrorCode = "HostInformationNotPresent"
+	InvalidDuplicateRow          TableErrorCode = "InvalidDuplicateRow"
+	InvalidInput                 TableErrorCode = "InvalidInput"
+	InvalidValueType             TableErrorCode = "InvalidValueType"
+	JSONFormatNotSupported       TableErrorCode = "JsonFormatNotSupported"
+	MethodNotAllowed             TableErrorCode = "MethodNotAllowed"
+	NotImplemented               TableErrorCode = "NotImplemented"
+	OutOfRangeInput              TableErrorCode = "OutOfRangeInput"
+	PropertiesNeedValue          TableErrorCode = "PropertiesNeedValue"
+	PropertyNameInvalid          TableErrorCode = "PropertyNameInvalid"
+	PropertyNameTooLong          TableErrorCode = "PropertyNameTooLong"
+	PropertyValueTooLarge        TableErrorCode = "PropertyValueTooLarge"
+	ResourceNotFound             TableErrorCode = "ResourceNotFound"
+	TableAlreadyExists           TableErrorCode = "TableAlreadyExists"
+	TableBeingDeleted            TableErrorCode = "TableBeingDeleted"
+	TableNotFound                TableErrorCode = "TableNotFound"
+	TooManyProperties            TableErrorCode = "TooManyProperties"
+	UpdateConditionNotSatisfied  TableErrorCode = "UpdateConditionNotSatisfied"
+	XMethodIncorrectCount        TableErrorCode = "XMethodIncorrectCount"
+	XMethodIncorrectValue        TableErrorCode = "XMethodIncorrectValue"
+	XMethodNotUsingPost          TableErrorCode = "XMethodNotUsingPost"
+)
+
+// PossibleTableErrorCodeValues returns a slice of all possible TableErrorCode values
+func PossibleTableErrorCodeValues() []TableErrorCode {
+	return []TableErrorCode{
+		DuplicatePropertiesSpecified,
+		EntityAlreadyExists,
+		EntityNotFound,
+		EntityTooLarge,
+		HostInformationNotPresent,
+		InvalidDuplicateRow,
+		InvalidInput,
+		InvalidValueType,
+		JSONFormatNotSupported,
+		MethodNotAllowed,
+		NotImplemented,
+		OutOfRangeInput,
+		PropertiesNeedValue,
+		PropertyNameInvalid,
+		PropertyNameTooLong,
+		PropertyValueTooLarge,
+		TableAlreadyExists,
+		TableBeingDeleted,
+		TableNotFound,
+		TooManyProperties,
+		UpdateConditionNotSatisfied,
+		XMethodIncorrectCount,
+		XMethodIncorrectValue,
+		XMethodNotUsingPost,
+	}
+}
+
+type oDataError struct {
+	Error oDataMessage `json:"odata.error"`
+}
+
+type oDataMessage struct {
+	Code TableErrorCode `json:"code"`
+}
+
+// parseErrorCode creates a azcore.ResponseError with the Code as one of the TableErrorCodes
+func parseErrorCode(svcErr error) error {
+	var httpErr *azcore.ResponseError
+	if errors.As(svcErr, &httpErr) {
+		body, err := ioutil.ReadAll(httpErr.RawResponse.Body)
```

Comment:
It's a bit unfortunate that we have to parse this again.  We have some handling for error code extraction in `azcore` [already](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/internal/shared/response_error.go#L47).  Might be better to fix that up to handle the `odata.error` case.

Reply:
I added this PR to add the case: https://github.com/Azure/azure-sdk-for-go/pull/17579

If that PR looks good, this will be much more simplified